@charleygros
Ok so I removed `clean_metadata`, which works for training but fails for `test_segment_volume` with this error:
```
ivadomed/loader/loader.py:760: in __getitem__
    stack_gt, metadata_gt = self.transform(sample=seg_pair_slice["gt"],
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ivadomed.transforms.Compose object at 0x7f1f9af3fdf0>, sample = None, metadata = None, data_type = 'gt'

    def __call__(self, sample, metadata, data_type='im'):
        print("print metadata:", metadata)
>       if self.transform[data_type] is None or len(metadata) == 0:
E       TypeError: object of type 'NoneType' has no len()

ivadomed/transforms.py:147: TypeError
```

I think this is because `clean_metadata` was returning an empty list, and now the metadata is just `None`.

We could either assign an empty list to the metadata like this in the loader [here](https://github.com/ivadomed/ivadomed/blob/32b911b335fcb36219be90c44c6700ccfa3989f4/ivadomed/loader/loader.py#L737):
`metadata_input = seg_pair_slice['input_metadata'] if seg_pair_slice['input_metadata'] is not None else []`

or change the condition in the call of Compose in transforms.py [here](https://github.com/ivadomed/ivadomed/blob/32b911b335fcb36219be90c44c6700ccfa3989f4/ivadomed/transforms.py#L146):
from `if self.transform[data_type] is None or len(metadata) == 0:` to `if self.transform[data_type] is None or not metadata:`

I'm not sure what are the implications of this and which solutions would be best. Let me know what you think.

--- a/ivadomed/loader/loader.py
+++ b/ivadomed/loader/loader.py
@@ -735,9 +735,9 @@ class MRI2DSegmentationDataset(Dataset):
                 seg_pair_slice['gt'][idx_class] = seg_pair_slice['gt'][idx_class][idx_rater]
                 seg_pair_slice['gt_metadata'][idx_class] = seg_pair_slice['gt_metadata'][idx_class][idx_rater]
 
-        metadata_input = seg_pair_slice['input_metadata']
-        metadata_roi = roi_pair_slice['gt_metadata']
-        metadata_gt = seg_pair_slice['gt_metadata']
+        metadata_input = seg_pair_slice['input_metadata'] if seg_pair_slice['input_metadata'] is not None else []
+        metadata_roi = roi_pair_slice['gt_metadata'] if roi_pair_slice['gt_metadata'] is not None else []
+        metadata_gt = seg_pair_slice['gt_metadata'] if seg_pair_slice['gt_metadata'] is not None else []
 
         # Run transforms on ROI
         # ROI goes first because params of ROICrop are needed for the followings
@@ -941,8 +941,8 @@ class MRI3DSubVolumeSegmentationDataset(Dataset):
                 seg_pair['gt'][idx_class] = seg_pair['gt'][idx_class][idx_rater]
                 seg_pair['gt_metadata'][idx_class] = seg_pair['gt_metadata'][idx_class][idx_rater]
 
-        metadata_input = seg_pair['input_metadata']
-        metadata_gt = seg_pair['gt_metadata']
+        metadata_input = seg_pair['input_metadata'] if seg_pair['input_metadata'] is not None else []
+        metadata_gt = seg_pair['gt_metadata'] if seg_pair['gt_metadata'] is not None else []
 
         # Run transforms on images
         stack_input, metadata_input = self.transform(sample=seg_pair['input'],

